apps/04TowersOfHanoi.js

```diff
@@ -38,119 +46,87 @@ function movePiece(startStack, endStack) {
 
 // make this work
 function isLegal(startStack, endStack) {
-    if (stacks[startStack] === undefined  ||  stacks[endStack] === undefined) {
-      if (stacks[startStack] === 'setwin'  ||  stacks[endStack] === 'setwin') {
-        stacks = setWin;
-        return true;
-      }
-      else {
+    var startStackArray = stacks[startStack];
+    var endStackArray = stacks[endStack];
+    if (startStackArray === undefined || endStackArray === undefined) {
         console.log('whoops, try again, this time enter a valid input (a, b, c):');
         return false;
-      }
     }
-    // if (stacks[startStack] === 'setwin'  ||  stacks[endStack] === 'setwin') {
-    //     if (stacks[startStack] === undefined  ||  stacks[endStack] === undefined) {
-    //       console.log('whoops, try again, this time enter a valid input (a, b, c):');
-    //       return false;
-    //     }
-    //     else {
-    //     stacks = setWin;
-    //     getPrompt();
-    //   }
-    // }
-    if (stacks[startStack].length === 0) {
-      return false;
-    }
-
-    if (stacks[endStack].length === 0) {
-      return true;
+    if (startStackArray.length === 0) {
+        return false;
     }
-    if (stacks[startStack].slice(-1) <= stacks[endStack].slice(-1)) {
-      return true;
+    if (endStackArray.length === 0) {
+        return true;
     }
-    // if (stacks[startStack][stacks[startStack].length - 1] <= stacks[endStack][stacks[endStack].length - 1])  {
-    //   return true;
-    // }
-    else {
-      console.log("Whoops, try again. You cannot move a bigger piece onto a smaller piece.");
-      return false;
+    if (startStackArray.slice(-1) <= endStackArray.slice(-1)) {
+        return true;
+    } else {
+        console.log("Whoops, try again. You cannot move a bigger piece onto a smaller piece.");
+        return false;
     }
 }
 
 function checkForWin() {
-  if (stacks.b.length >= 4 || stacks.c.length >= 4)
-  {
-    console.log('you won!');
-    return true;
-  }
-  else {
-    return false;
-  }
+    var bStack = stacks.b;
+    var cStack = stacks.c;
+    if ((bStack.length >= 4 || cStack.length >= 4)) {
+        console.log('you won!');
+        console.log("You got it in " + numMoves + " moves. Think you can do better?");
+        resetStacks();
+        playAgain();
```

Comment:
This is great!!!! When you're feeling up to it, another challenge would be to have the option for the computer to complete the game by itself and show its moves.